lib/extensions.js

```diff
@@ -252,6 +256,10 @@ module.exports = function (realmConstructor) {
                 if (cancelled) {
                   return;
                 }
+                // Clear the fallback timeOut if it has been started
+                if (timeoutId) {
+                  clearTimeout(timeoutId);
+                }
```

Comment:
In an ideal scenario I would love this to go into a `finally(() => { /
 ...*/ })` on the [`openPromise`](https://github.com/realm/realm-js/blob/andrew/timeOutBehavior-fix/lib/extensions.js#L274) - but let's save that for another PR 😄

Reply:
Sounds good.  Someday we should just rewrite this entirely.  I'm also kind of unhappy about `timeOut` vs `timeout`.  Maybe a breaking change ticket is in order